src/pages/lib/auth/password_management/q/platform/[platform].mdx

```diff
@@ -1,8 +1,202 @@
+
+## Allow my users to replace a forgotten password
+
+Your users may want to retrieve access to their account and replace a forgotten password. They can either perform self-service and reset the password after verifying ownership of the account, or request an Admin to reset the password for their account.
+
+### Change to a new password via self-service
+
+You can enable self-service by using the `forgotPassword` method. This will send a confirmation code to your user's email or phone number (depending on which attributes are verified in your Cognito user pool) and they can then enter that confirmation code with a new password using the `forgotPasswordSubmit` method. This will allow the user to reset their password and sign in with the new password.
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+
+// Send confirmation code to user's email
+async function forgotPassword(username: string) {
+  try {
+    const data = await Auth.forgotPassword(username);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+};
+
+// Collect confirmation code and new password
+async function forgotPasswordSubmit(username: string, code: string, newPassword: string) {
+  try {
+    const data = await Auth.forgotPasswordSubmit(username, code, newPassword);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+};
+```
+</Block>
+<Block name="JavaScript">
+
+```javascript
+import { Auth } from 'aws-amplify';
+
+// Send confirmation code to user's email
+async function forgotPassword(username) {
+  try {
+    const data = await Auth.forgotPassword(username);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+} 
+
+// Collect confirmation code and new password
+async function forgotPasswordSubmit(username, code, newPassword) {
+  try {
+    const data = await Auth.forgotPasswordSubmit(username, code, newPassword);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+}
+```
+</Block>
+</BlockSwitcher>
+
+### Change to a new password after Admin reset
+
+In the case where the user's account password needs to be reset by an Admin, a confirmation code will be sent to your user's email or phone number (depending on which attributes are verified in your Cognito user pool) as soon as the reset is triggered. They can then enter that confirmation code with a new password using the `forgotPasswordSubmit` method. This will allow the user to reset their password and sign in with the new password.
+
+<Callout>
+
+**Note:** An Admin can reset a user's password by going into the Cognito Userpool, selecting the user, and choosing "Reset password" under the *Actions* dropdown. This can also be done programmatically using the Cognito API Action `AdminResetUserPassword`. Learn more about [resetting a users password as an Admin](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminResetUserPassword.html).
```

Comment:
**Note:** An Admin can reset a user's password by going into the Cognito Userpool console, selecting the user, and choosing "Reset password" under the *Actions* dropdown. This can also be done programmatically using the Cognito API Action `AdminResetUserPassword`. Learn more about [resetting a users password as an Admin](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminResetUserPassword.html).

Reply:
Updated